### PR TITLE
Add ranked stats page

### PR DESCRIPTION
**Pull request by bdamore**
modified add-ranked-stats-page

### CODE REVIEW

File: .jshintrc [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (luyang-gai) [inline .jshintrc]: what is this jshint config change doing, if we don't know, let's not commit

File: app/Services/lolapi.js [hunk 2]
```diff
@@ -38,18 +45,27 @@ angular.module("LeagueViewer").factory("lolapi", function($http) {
     });
   };
 
-  var getRankedStats = function(championId) {
-    return $http.get(baseUrl + "champion/" + championId)
+  // var getRankedStats = function(championId) {
```
Comment (luyanggai23) [inline app/Services/lolapi.js]: remove comments

File: app/Services/lolapi.js [hunk 3]
```diff
@@ -3,6 +3,13 @@ var app = angular.module("LeagueViewer");
 angular.module("LeagueViewer").factory("lolapi", function($http) {
   var baseUrl = "http://localhost:3000/";
 
+  var getSummoner1 = function(summonerName) {
```
Comment (luyanggai23) [inline app/Services/lolapi.js]: Let's come up with a better name to distinguish one vs. the other (getSummoner vs getSummoner1)

File: app/components/main/main-routes.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  'use strict';
+
+  angular.module('LeagueViewer')
+    .config(function ($stateProvider) {
+      $stateProvider
+          .state('main', {
+              url: '/main',
+              templateUrl: 'app/components/main/main.html',
+              controller: 'MainController'
+              // resolve: {
```
Comment (luyanggai23) [inline app/components/main/main-routes.js]: comments

File: app/components/main/main-routes.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
+(function () {
```
Comment (luyanggai23) [inline app/components/main/main-routes.js]: is this file being used?

File: app/components/match-history/match-history.html [hunk 6]
```diff
@@ -1,12 +1,15 @@
 <div class="match-history-content container-fluid">
 	<div class="summoner-bar" ng-if="haveResults">
+
 		<div class="general-summoner-info">
 			<div class="summoner-name">Summoner: {{summoner.summonerName}}</div>
+			<h4 class="blue"><a ng-href="#/home/ranked-stats/{{summoner.summonerName}}">View ranked stats</h4></a>
```
Comment (luyanggai23) [inline app/components/match-history/match-history.html]: let's not use ng-href and instead and an ng-click event that will call a function on the controller's scope, which will handle the transition. Similar to the 'More stats for this match' button.

File: app/components/ranked-stats/ranked-stats-ctrl.js [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+            getRankedInfo(summonerId);
+
+          },
+          function(failure) {
+            console.log('fail');
+          }
+        );
+      };
+
+      // var getSummonerId = function(summonerId) {  summId
```
Comment (luyanggai23) [inline app/components/ranked-stats/ranked-stats-ctrl.js]: comments

File: app/components/ranked-stats/ranked-stats-ctrl.js [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+          function(result) {
+            console.log('test');
+            $scope.summonerObject = result[summonerName];
+            summonerId = $scope.summonerObject.id;
+            // getSummonerId(); summId
+            getRankedInfo(summonerId);
+
+          },
+          function(failure) {
+            console.log('fail');
```
Comment (luyanggai23) [inline app/components/ranked-stats/ranked-stats-ctrl.js]: let's have this call an onError function, similar to what we do in match-history-ctrl

File: app/components/ranked-stats/ranked-stats-ctrl.js [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+        $scope.errorMessage = 'Could not get Summoner Info';
+      };
+
+      var getSummoner1 = function() {
+        lolapi.getSummoner1(summonerName).then(
+          function(result) {
+            console.log('test');
+            $scope.summonerObject = result[summonerName];
+            summonerId = $scope.summonerObject.id;
+            // getSummonerId(); summId
```
Comment (luyanggai23) [inline app/components/ranked-stats/ranked-stats-ctrl.js]: comments

File: backend/routes/index.js [hunk 10]
```diff
@@ -106,11 +106,30 @@ router.get('/summoner/:summonerId/matchHistory', function(req, res, next) {
 	});
 });
 
+//GET SUMMONER RANKED-STATS
```
Comment (luyanggai23) [inline backend/routes/index.js:109]: spacing?

File: app/components/match-history/match-history.html [hunk 11]
```diff
@@ -1,12 +1,18 @@
 <div class="match-history-content container-fluid">
 	<div class="summoner-bar" ng-if="haveResults">
+
 		<div class="general-summoner-info">
 			<div class="summoner-name">Summoner: {{summoner.summonerName}}</div>
+
+			<!-- <h4 class="blue"><a ng-href="#/home/ranked-stats/{{summoner.summonerName}}">View ranked stats</h4></a> -->
```
Comment (luyang-gai) [inline app/components/match-history/match-history.html:7]: remove comment

**Pull request merged**